### PR TITLE
Resolve StrictMode violation in App Check.

### PR DESCRIPTION
**Pull request by rosalyntan**
Resolve a StrictMode violation in App Check by moving the retrieval of the stored App Check token out of the initialization path.

### CODE REVIEW

**Review by emilypgoogle — APPROVED** [review]

File: appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java [hunk 1]
```diff
@@ -146,44 +176,58 @@ public void removeAppCheckListener(@NonNull AppCheckListener listener) {
...
-                // If the token exchange failed, return a dummy token for integrators to attach in
-                // their headers.
-                return Tasks.forResult(
-                    DefaultAppCheckTokenResult.constructFromError(
-                        new FirebaseException(
-                            task.getException().getMessage(), task.getException())));
-              }
-            });
+    return retrieveStoredTokenTask.continueWithTask(
+        new Continuation<Void, Task<AppCheckTokenResult>>() {
```
Comment (vkryachko) [inline appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java:180]: nit: can you switch to lambda syntax here and below?

**Review by vkryachko — APPROVED** [review]

File: appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java (hunk 1 above)
Comment (rosalyntan) [reply appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/internal/DefaultFirebaseAppCheck.java]: Done.

**Comment by rosalyntan** [comment]
@vkryachko are the Firebase Storage unit tests known to be flaky? They all pass every time when I run them locally, but they seem to be failing in the presubmits here (specifically `UploadTest`). I got the unit test presubmit to run on previous commits after re-running them several times, but the `Health Metrics / Coverage` tests are consistently failing.

**Comment by vkryachko** [comment]
Yes, storage has been really flaky lately, Health Metrics are non-blocking, so no need to worry about them

**Pull request merged**